Source/Core/DolphinQt/Config/Graphics/GeneralWidget.cpp

```diff
@@ -375,3 +393,141 @@ void GeneralWidget::OnBackendChanged(const QString& backend_name)
+      Config::GFX_SHOW_FPS,
+      Config::GFX_BITRATE_KBPS,
+      Config::GFX_ENHANCE_MAX_ANISOTROPY,
+      Config::GFX_HACK_BBOX_ENABLE,
+      Config::GFX_HACK_VERTEX_ROUNDING,
+      Config::GFX_HACK_VI_SKIP,
+      Config::GFX_ENHANCE_FORCE_TEXTURE_FILTERING,
+      Config::GFX_CC_GAME_COLOR_SPACE,
+      Config::GFX_STEREO_DEPTH,
+      Config::GFX_DUMP_EFB_TARGET,
+      Config::GFX_HACK_SKIP_XFB_COPY_TO_RAM,
+      Config::GFX_SSAA,
+      Config::GFX_PERF_SAMP_WINDOW,
+      Config::MAIN_PRECISION_FRAME_TIMING,
+      Config::GFX_SHOW_VPS,
+      Config::GFX_SHADER_COMPILATION_MODE,
+      Config::GFX_HACK_SKIP_DUPLICATE_XFBS,
+      Config::GFX_SAVE_TEXTURE_CACHE_TO_STATE,
+      Config::GFX_MODS_ENABLE,
+      Config::GFX_EFB_SCALE,
+      Config::GFX_DUMP_BASE_TEXTURES,
+      Config::GFX_SHOW_NETPLAY_MESSAGES,
+      Config::GFX_FAST_DEPTH_CALC,
+      Config::GFX_MAX_EFB_SCALE,
+      Config::GFX_CC_HDR_PAPER_WHITE_NITS,
+      Config::GFX_SAFE_TEXTURE_CACHE_COLOR_SAMPLES,
+      Config::GFX_ADAPTER,
+      Config::GFX_DUMP_MIP_TEXTURES,
+      Config::GFX_CACHE_HIRES_TEXTURES,
+      Config::GFX_BORDERLESS_FULLSCREEN,
+      Config::GFX_MSAA,
+      Config::GFX_ENABLE_VALIDATION_LAYER,
+      Config::GFX_STEREO_CONVERGENCE,
+      Config::GFX_HIRES_TEXTURES,
+      Config::GFX_CC_GAME_GAMMA,
+      Config::GFX_HACK_IMMEDIATE_XFB,
+      Config::GFX_PNG_COMPRESSION_LEVEL,
+      Config::GFX_ENHANCE_OUTPUT_RESAMPLING,
+      Config::GFX_VSYNC,
+      Config::GFX_CROP,
+      Config::GFX_CUSTOM_ASPECT_RATIO_HEIGHT,
+      Config::GFX_HACK_DISABLE_COPY_TO_VRAM,
+      Config::GFX_HACK_COPY_EFB_SCALED,
+  };
+  // Array generated using this Python Script:
+  // import re, os
+  // k = set()
+  // for r, _, f in os.walk("Graphics"):
+  //   for file in f:
+  //     if file.endswith((".cpp", ".h")):
+  //       with open(os.path.join(r, file), errors="ignore") as x:
+  //         k |= set(re.findall(r'\bConfig::([A-Z0-9_]+)\b', x.read()))
+  // s = list(set(sorted(k)))
+  // print(f"const std::array<InfoVariant, {len(s)}> CONFIG_KEYS = {{")
+  // for i in range(0, len(s), 4):
+  //   print("        " + ", ".join(f"Config::{x}" for x in s[i:i+4]) + ",")
+  // print("}\n")
+
+  auto response = ModalMessageBox::question(
+      this, tr("Reset Dolphin"),
+      tr("Are you sure you want to restore all Dolphin graphics settings to their default values? "
+         "This action is irreversible!\n"
+         "Any customizations you've made will be permanently lost.\n\n"
+         "Do you wish to continue?"),
+      QMessageBox::StandardButton::Yes | QMessageBox::StandardButton::Cancel,
+      QMessageBox::StandardButton::Cancel);
+
+  if (response != QMessageBox::StandardButton::Yes)
+    return;
+
+  for (const auto& info_variant : CONFIG_KEYS)
+  {
+    std::visit(
+        [](const auto& info) {
+          const auto default_value = info.GetDefaultValue();
+
+          Config::SetBaseOrCurrent(info, default_value);
```

Comment:
It might be more appropriate to delete the key rather than explicitly set the default value?